crates/driver/src/infra/config/solvers.rs

```diff
@@ -0,0 +1,47 @@
+use {
+    crate::{domain::eth, infra::solver, util::serialize},
+    serde::Deserialize,
+    serde_with::serde_as,
+    std::{fs, path::Path},
+};
+
+/// Load the solver configuration from a YAML file. Despite being an I/O
+/// operation, this is done at startup, so it doesn't need to be async.
+pub fn load(path: &Path) -> Vec<solver::Config> {
+    let data = fs::read(path).unwrap_or_else(|_| panic!("I/O error while reading {path:?}"));
+    let config: Config = serde_yaml::from_slice(&data)
+        .unwrap_or_else(|_| panic!("YAML syntax error while reading {path:?}"));
```

Comment:
Why panic instead of returning an error? The answer is probably that this function is only used at startup where we would panic anyway. Still, the overhead of converting this to result is probably very small and then I'd rather unwrap in main so it is clear that that is what's going on. If the panic is kept inside here then it should be documented.

Reply:
> Why panic instead of returning an error?

For things like loading configurations, I think it is fine to panic within the function. My reasoning is that:
- You no longer need to pass the configuration around as any code can `Config::load()` and not need to panic internally.
- Invalid configuration is an error case where I don't see any code needing to handle the error case (if its wrong, there is nothing to do).

> If the panic is kept inside here then it should be documented.

Agreed